pandas/tests/frame/test_arithmetic.py

```diff
@@ -823,3 +823,13 @@ def test_align_frame(self):
         half = ts[::2]
         result = ts + half.take(np.random.permutation(len(half)))
         tm.assert_frame_equal(result, expected)
+
+
+def test_pow_with_realignment():
+    # GH#32685 pow has special semantics for operating with null values
+    left = pd.DataFrame({"A": [0, 1, 2]})
+    right = pd.DataFrame(index=[0, 1, 2])
+
+    result = left ** right
```

Comment:
This tests `x ** nan  == x`.

Can you add a test for `nan ** 0 == 1`?

Perhaps parametrize over series & frame ops?